### DISCUSSION TITLE
Without Codacy's `Remark-lint` triggering `[no-undefined-references]` `[no-shortcut-reference-link]` notices, how to use Markdown footnotes?

### DISCUSSION THREAD

**Discussion by SwuduSusuwu** [remarkjs/remark#1392] [Q&A]
Found https://github.com/remarkjs/remark-lint/issues/234, which just says that you have to update `.remarkrc` to allow use of _Pandoc_ footnotes.

But the problem is with `Remark-lint` through _GitHub_'s `Codacy` scan tool (which does not have `.remarkrc` to configure, just [.github/workflows/codacy.yml](https://github.com/SwuduSusuwu/SubStack/blob/trunk/.github/workflows/codacy.yml)).

Must disable _Codacy_'s  whole `Remark-lint` just to use footnotes? Or is there some other form of footnotes which `Remark-lint` allows (just not _Pandoc_ footnotes)?

List of `Remark-lint` errors is at https://github.com/SwuduSusuwu/SubStack/runs/34401539110, the exact commit which triggered this is https://github.com/SwuduSusuwu/SubStack/commit/188d87fdc2b1496238f4e792a003cb328c3f423f

https://github.com/remarkjs/remark-lint/issues/234 says that _Pandoc_ block footnotes should not trigger `no-undefined-references`, so perhaps this is not a duplicate.

**Comment by SwuduSusuwu** [comment]
Is this problem similar to https://github.com/sindresorhus/awesome-lint/issues/160 ? Is it that derivatives of `Remark-lint` (which do not expose their own `.remarkrc`) have no solution?

**Comment by SwuduSusuwu** [comment]
https://github.com/codacy/codacy-remark-lint/issues redirects to https://github.com/codacy/codacy-remark-lint/pulls (or would have posted them an issue for this.)

**Comment by SwuduSusuwu** [comment]
Am just aware of the "_Pandoc_" style of footnotes.
Asked _Assistant_ about other versions of footnotes, but all it gave was the `[^<number>]` version of footnotes.
Which version of footnotes does `Remark-lint` allow you to use?

**Comment by ChristianMurphy** [comment]
> Found https://github.com/remarkjs/remark-lint/issues/234, which just says that you have to update .remarkrc to allow use of Pandoc footnotes.

That issue is 4 years old and refers to code multiple major versions ago.
The new version support GFM style https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/

> Must disable Codacy's whole Remark-lint just to use footnotes? Or is there some other form of footnotes which Remark-lint allows (just not Pandoc footnotes)?

Take a moment to read https://unifiedjs.com/learn/
`remark-lint` works on syntax trees, which allows you to add or remove syntax features without needing to disable lint rules.
If you add a parser that supports your flavor of footnotes the rules will work.

> Is this problem similar to https://github.com/sindresorhus/awesome-lint/issues/160 ?

Yes in the sense of the last comment.
You need to define the extensions to markdown you want to support in a config.

> Is it that derivatives of Remark-lint (which do not expose their own .remarkrc) have no solution?

If codecy disables or overrides your remark configuration that would indeed be problematic for supporting footnotes.
But that is an issue with codecy not here.

**Reply by SwuduSusuwu** [reply]
> The new version support GFM style https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/

That shows the _Pandoc_ syntax of block footnotes (used in the 4 year old issue, and which triggered _Codacy_'s `Remark-lint`.)


> 
> > Must disable Codacy's whole Remark-lint just to use footnotes? Or is there some other form of footnotes which Remark-lint allows (just not Pandoc footnotes)?
> 
> Take a moment to read https://unifiedjs.com/learn/ `remark-lint` works on syntax trees, which allows you to add or remove syntax features without needing to disable lint rules. If you add a parser that supports your flavor of footnotes the rules will work.

It sounds as if that just has use for local `Remark-lint` installs. The question is; for cloud tools (which use the default `Remark-lint` rules,) how to use _Markdown_ footnotes?

> > Is this problem similar to [sindresorhus/awesome-lint#160](https://github.com/sindresorhus/awesome-lint/issues/160) ?
> 
> Yes in the sense of the last comment. You need to define the extensions to markdown you want to support in a config.
> 
> > Is it that derivatives of Remark-lint (which do not expose their own .remarkrc) have no solution?
> 
> If codecy disables or overrides your remark configuration that would indeed be problematic for supporting footnotes. But that is an issue with codecy not here.

_Codacy_ uses the default `Remark-lint` rules. it does not offer options to change the configuration. The issue is that `Remark-lint`'s default rules do not allow footnotes, although most instances of `Markdown` for the last 4 years have supported footnotes.

**Reply by ChristianMurphy** [reply]
> Codacy uses the default Remark-lint rules. it does not offer options to change the configuration.

Raise this with codecy

> The issue is that Remark-lint's default rules do not allow footnotes

No, it supports footnotes.
Their/your configuration has them disabled.

> The question is; for cloud tools (which use the default Remark-lint rules,) how to use Markdown footnotes?

Have the cloud tool support configuration so you can enable or disable things as you please.
